README.md

```diff
@@ -1,63 +1,91 @@
+| SignatureSender (Proxy) |  [0x1c86A81D3CDD897aFdcA62a9b7219a39Aef7910B](https://calibration.filfox.info/en/address/0x1c86A81D3CDD897aFdcA62a9b7219a39Aef7910B)  |
+| SignatureSender (Impl) | [0x1790de5a9fBA748DCAf05e3a1755Cf1DD6b9B0F8](https://calibration.filfox.info/en/address/0x1790de5a9fBA748DCAf05e3a1755Cf1DD6b9B0F8)   |
+| SignatureSchemeAddressProvider |  [0xD2b5084E68230D609AEaAe5E4cF7df9ebDd6375A](https://calibration.filfox.info/en/address/0xD2b5084E68230D609AEaAe5E4cF7df9ebDd6375A)  |
+
+## Quick Start  
+### Installation
+To get started, install the randomness-solidity library in your smart contract project using your preferred development tool.
 
-##### Foundry 
-```sh
-$ forge install randa-mu/randomness-solidity
+**Hardhat (npm)**
+```bash
+npm install randomness-solidity
+```  
+**Foundry**
+```bash
+forge install randa-mu/randomness-solidity
 ```
 
-#### Importing
+### How to use
 
-To use this library in your project, import the required files into your contract and use the proxy contract address for RandomnessSender in the constructor as the randomnessSender address parameter:
+1. **Import the library**
 
-```solidity
-// Import the abstract RandomnessReceiverBase contract for creating randomness requests and handling randomness callbacks
-import {RandomnessReceiverBase} from "../RandomnessReceiverBase.sol";
-```
+    Start by importing the `RandomnessReceiverBase.sol` abstract contract into your smart contract. This contract provides the interface for making randomness requests and handling callbacks
 
-#### Example Usage
+    ```solidity
+    // Import the abstract RandomnessReceiverBase contract for creating randomness requests and handling randomness callbacks
+    import {RandomnessReceiverBase} from "../RandomnessReceiverBase.sol";
+    ```
 
-```solidity
-// SPDX-License-Identifier: MIT
-pragma solidity ^0.8;
+2. **Extend the  `RandomnessReceiverBase` contract**
 
-import {RandomnessReceiverBase} from "../RandomnessReceiverBase.sol";
+   To use the library, your contract must inherit from `RandomnessReceiverBase` and specify the deployed `RandomnessSender` contract address from your desired [network](#support-network) in the constructor. 
 
-contract MockRandomnessReceiver is RandomnessReceiverBase {
-    bytes32 public randomness;
-    uint256 public requestId;
+    ```solidity
+    contract diceRoller is RandomnessReceiverBase {
+        constructor(address randomnessSender) RandomnessReceiverBase(randomnessSender) {}
+        ...
+    }
+    ```
 
-    constructor(address randomnessSender) RandomnessReceiverBase(randomnessSender) {}
+3. **Request Randomness**
 
+    Use the `requestRandomness()` function to send a randomness request to the Dcipher network. This request will be forwarded to the pre-deployed `RandomnessSender` contract.
+
+    The function returns a `requestId`, which should be stored to verify the response when randomness is delivered.
```

Comment:
the response is automatically verified which is worth highlighting

Reply:
Good point, I will highlight it on the 4th step when the random value is returned.